packages/@lwc/shared/src/style.ts

```diff
@@ -5,10 +5,39 @@
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
  */
 
+import { isArray } from './language';
+
 export const IMPORTANT_FLAG = /\s*!\s*important\s*$/i;
 const DECLARATION_DELIMITER = /;(?![^(]*\))/g;
 const PROPERTY_DELIMITER = /:(.+)/s; // `/s` (dotAll) required to match styles across newlines, e.g. `color: \n red;`
 
+/**
+ * Function producing style based on a host and a shadow selector. This function is invoked by
+ * the engine with different values depending on the mode that the component is running on.
+ */
+export type Stylesheet = {
+    /**
+     * Function taking a stylesheet token (string), whether to render actual or "scoped" `:host()` pseudo-classes,
+     * and whether to render actual or synthetic `:dir()` pseudo-classes and returning a CSS string.
+     * This function signature should _not_ be considered a stable API surface - it is internal to the LWC engine.
```

Comment:
Well if you have `Stylesheets` then you should presumably also have `Stylesheet`. But I don't have a strong opinion on this since I think it's way too easy to break types across releases, so it just kind of comes with the territory.